Discord Tools/Integrations.md

```diff
@@ -0,0 +1,117 @@
+
+        // Checks if there's an ID, if there is, convert into a mention.
+        if (bonusFeatures.convert2Mention == 'ON' && !isNaN(item.value) && item.value.length == 18) item.value = `<@!${item.value}>`;
+
+        // Checks if type is text, or default to embed.
+        if (bonusFeatures.convert2Link == 'ON' && type.toLowerCase() !== 'text') {
+            // Validates the link, if it's true, make it a hyperlink for embed.
+            if (linkValidate.test(item.value)) item.value = `[${item.value}](${item.value})`;
+        } else {
+            // Validates the link, if it's true, make it not-embed for text.
+            if (bonusFeatures.convert2Link == 'ON' && linkValidate.test(item.value)) item.value = `<${item.value}>`;
+        }
+
+        return [`**${item.name}**`, `${item.value}`].join("\n");
+    }
+}
+
+if (items.map(data).toString().length + shortDescription.length > 1999) throw "Discord limit reached. Please add limits to your questions!";
+
+function plainText(e) {
+
+    // A webhook construct, which sets up the correct formatting for sending to Discord.
+    const text = {
+        "method": "post",
+        "headers": { "Content-Type": "application/json" },
+        "muteHttpExceptions": true,
+        "payload": JSON.stringify({
+            "content": `${mention ? mention : ''}${title ? `**${title}**` : `**${form.getTitle()}**`}\n\n${shortDescription ? `${shortDescription}\n\n${items.map(data).join('\n\n')}` : items.map(data).join('\n\n')}`
+        }),
+    };
+
+    // We now loop through our webhooks and send them one by one to the respectful channels.
+    for (var i = 0; i < webhooks.length; i++) { UrlFetchApp.fetch(webhooks[i], text); };
+}
+
+function embedText(e) {
+
+    // A webhook embed construct, which sets up the correct formatting for sending to Discord.
+    const embed = {
+        "method": "post",
+        "headers": { "Content-Type": "application/json" },
+        "muteHttpExceptions": true,
+        "payload": JSON.stringify({
+            "content": mention ? mention : '',
+            "embeds": [{
+                "title": title ? title : form.getTitle(), // Either the set title or the forms title.
+                "description": shortDescription ? `${shortDescription}\n\n${items.map(data).join('\n\n')}` : items.map(data).join('\n\n'), // Either the desc or just the res.
+                "thumbnail": { url: avatarImage ? encodeURI(avatarImage) : null }, // The tiny image in the right of the embed
+                "color": colour ? parseInt(colour.substr(1), 16) : Math.floor(Math.random() * 16777215), // Either the set colour or random.
+                "timestamp": new Date().toISOString() // Today's date.
+            }]
+        }),
+    };
+
+    // We now loop through our webhooks and send them one by one to the respectful channels.
+    for (var i = 0; i < webhooks.length; i++) { UrlFetchApp.fetch(webhooks[i], embed); };
+}```
```

Comment:
Credits go to an unknown source for writing this integration that is used to transfer google form responses to discord